web/src/compose_fade_helper.ts

```diff
@@ -1,27 +1,36 @@
+import assert from "minimalistic-assert";
+
 import * as stream_data from "./stream_data";
 import * as sub_store from "./sub_store";
 import * as util from "./util";
+import type {DirectRecipient, StreamRecipient} from "./util";
+
+type Recipient = DirectRecipient | StreamRecipient;
 
-let focused_recipient;
+let focused_recipient: Recipient | undefined;
 
-export function should_fade_message(message) {
+export function should_fade_message(message: Recipient): boolean {
```

Comment:
This is called with a message, not with a recipient, so we need to fix `Message` to be a supertype of `Recipient`. Fixed in #26812.